lib/puma/request.rb

```diff
@@ -22,7 +22,7 @@ module Request # :nodoc:
     IO_BODY_MAX = 1_024 * 64
 
     # max size for io_buffer, force write when exceeded
-    IO_BUFFER_LEN_MAX = 1_024 * 1_024 * 4
```

Comment:
This is the change to close #3000 right? We should separate these.

Reply:
Done.  I'll create another PR, setting this as a DSL option called `size_to_first_byte`, as it affects TTFB.